apps/fxc-front/src/app/components/2d/path-element.ts

```diff
@@ -254,6 +250,63 @@ export class PathElement extends connect(store)(LitElement) {
     this.postScoreToHost(score);
   }
 
+  private computeScore(points: LatLon[]): Score {
+    const track: ScoringTrack = {
+      points: points.map((point, i) => {
+        return {
+          ...point,
+          alt: 0,
+          timeSec: i * 60,
+        };
+      }),
+      minTimeSec: new Date().getTime() / 1000,
+    };
+    const result = optimize({ track }, this.getLeague()).next().value;
+    const score = new Score({
+      circuit: this.getCircuitType(result),
+      distance: result.lengthKm * 1000,
+      multiplier: result.multiplier,
+      closingRadius: result.closingRadius ? result.closingRadius * 1000 : null,
+      indexes: result.solutionIndices,
+    });
+    // force the score as computed because of an unwanted side effect in constructor.
+    score.forcePoints(result.score);
+    return score;
+  }
+
+  private getLeague(): ScoringRules {
+    switch (this.league as LeagueCode) {
+      case 'czl':
+        return ScoringRules.CzechLocal;
+      case 'cze':
+        return ScoringRules.CzechEuropean;
+      case 'czo':
+        return ScoringRules.CzechOutsideEurope;
+      case 'fr':
+        return ScoringRules.FederationFrancaiseVolLibre;
+      case 'leo':
+        return ScoringRules.Leonardo;
+      case 'nor':
+        return ScoringRules.Norway;
+      case 'ukc':
+        return ScoringRules.UnitedKingdomClub;
+      case 'uki':
+        return ScoringRules.UnitedKingdomInternational;
+      case 'ukn':
+        return ScoringRules.UnitedKingdomNational;
+      case 'xc':
+        return ScoringRules.XContest;
+      case 'xcppg':
+        return ScoringRules.XContestPPG;
+      case 'wxc':
+        return ScoringRules.WorldXC;
+    }
+  }
+
+  private getCircuitType(result: OptimizationResult) {
+    return result.circuit as unknown as CircuitType;
```

Comment:
Same here you create a function the `logic` folder and probably pass an `OptimizationResult.result`

Reply:
code moved